### DISCUSSION TITLE
Need sample code for token swap using typescript sdk

### DISCUSSION THREAD

**Discussion by dsa012** [aptos-labs/aptos-developer-discussions#228] [Questions] [answered]
### Describe your question in detail.

Hi, i am new to aptos and i was not able to find code examples for token swap that uses aptos typescript sdk.

### What have you tried or looked at?  Or how can we reproduce the error?

I searched aptos docs and google, but didn't find something helpful

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by kshitijc1506** [comment, marked as answer]
This should help - https://github.com/aptos-labs/aptos-developer-discussions/discussions/111 (check out the simulation section). 

The discussion above tells you how to get the quote for a swap, in order to execute the swap you can simply build, sign and submit (instead of simulating) the transaction. Here's the [documentation](https://aptos.dev/sdks/ts-sdk/transaction-builder#submit-transaction) on how you can do that. 

Some DEXs have their own SDK you can use too.
